### ISSUE TITLE
thread 'block2' panicked at 'assertion failed - v0.8.0

### ISSUE THREAD

**Issue by bobdobs** [input-output-hk/jormungandr#1351]
**Describe the bug**
The error does not crash the node and it continues to run. No new blocks get added. 

**Mandatory Information**
1. `jcli --full-version` output;
`jcli 0.8.0 (HEAD-7868354, release, linux [x86_64]) - [rustc 1.39.0 (4560ea788 2019-11-04)]`
2. `jormungandr --full-version` output;
`jormungandr 0.8.0 (HEAD-7868354, release, linux [x86_64]) - [rustc 1.39.0 (4560ea788 2019-11-04)]`

**To Reproduce**
1. Start the pool/node: 
`jormungandr --genesis-block-hash 65a9b15f82619fffd5a7571fdbf973a18480e9acf1d2fddeb606ebb53ecca839 --config config.yaml --secret ./accounts/pool/node_secret.yaml`

**Expected behavior**
Did not expect panic. I was running v0.8.0-rc2 for a week, but since upgrading to rc-10 I have experienced problems. 

**Additional context**
```
thread 'block2' panicked at 'assertion failed: !self.candidate_map.contains_key(&root_hash)', jormungandr/src/blockchain/candidate.rs:468:17
stack backtrace:
   0:     0x5626ff3ed67c - backtrace::backtrace::libunwind::trace::hda41dbcdfba36aa0
                               at /cargo/registry/src/github.com-1ecc6299db9ec823/backtrace-0.3.37/src/backtrace/libunwind.rs:88
   1:     0x5626ff3ed67c - backtrace::backtrace::trace_unsynchronized::h1a8d6e1f8cb3f5d4
                               at /cargo/registry/src/github.com-1ecc6299db9ec823/backtrace-0.3.37/src/backtrace/mod.rs:66
   2:     0x5626ff3ed67c - std::sys_common::backtrace::_print_fmt::h610c4127487e10da
                               at src/libstd/sys_common/backtrace.rs:76
   3:     0x5626ff3ed67c - <std::sys_common::backtrace::_print::DisplayBacktrace as core::fmt::Display>::fmt::h0722dc552e01bd1d
                               at src/libstd/sys_common/backtrace.rs:60
   4:     0x5626fed737ac - core::fmt::write::h01edf6dd68a42c9c
                               at src/libcore/fmt/mod.rs:1030
   5:     0x5626ff3ece96 - std::io::Write::write_fmt::hf15985f193f03c04
                               at src/libstd/io/mod.rs:1412
   6:     0x5626ff3ecb10 - std::sys_common::backtrace::_print::hd8d5d08a1795e743
                               at src/libstd/sys_common/backtrace.rs:64
   7:     0x5626ff3ecb10 - std::sys_common::backtrace::print::hf89a79e3921a2366
                               at src/libstd/sys_common/backtrace.rs:49
   8:     0x5626ff3ecb10 - std::panicking::default_hook::{{closure}}::h3a8f42beb3bb8ae3
                               at src/libstd/panicking.rs:196
   9:     0x5626ff3ec25f - std::panicking::default_hook::h8f803b0bc31a5c37
                               at src/libstd/panicking.rs:210
  10:     0x5626ff3ec25f - std::panicking::rust_panic_with_hook::h825f041245da8739
                               at src/libstd/panicking.rs:473
  11:     0x5626ff26d6a4 - std::panicking::begin_panic::he11d6503ce9e2932
  12:     0x5626ff2d6c49 - jormungandr::blockchain::candidate::CandidateForestThickets::remove_root::h0ef473e78ff693a0
  13:     0x5626ff11f706 - <futures::future::map::Map<A,F> as futures::future::Future>::poll::h616d9503867681a4
  14:     0x5626ff045336 - <futures::future::either::Either<A,B> as futures::future::Future>::poll::hd7a44c1fc2d1ac12
  15:     0x5626ff0320e1 - <futures::future::either::Either<A,B> as futures::future::Future>::poll::h7bc7299059a5195a
  16:     0x5626ff2a2f71 - <futures::future::and_then::AndThen<A,B,F> as futures::future::Future>::poll::h7fed8b072e0e14cb
  17:     0x5626febfb1b3 - <futures::future::map_err::MapErr<A,F> as futures::future::Future>::poll::h8ad8297b85ea8819
  18:     0x5626ff43ef1e - futures::task_impl::std::set::h1bd778fe2d164d41
  19:     0x5626ff43af1c - tokio_threadpool::worker::Worker::run_task::h1508e513e3a67a63
  20:     0x5626ff4391c0 - tokio_threadpool::worker::Worker::run::h0448d4fb460195e5
  21:     0x5626ff41c00c - tokio_reactor::with_default::h71ab6fd20a390252
  22:     0x5626ff41eee5 - tokio::runtime::threadpool::builder::Builder::build::{{closure}}::h3d24c038b3dabae8
  23:     0x5626ff4373db - std::sys_common::backtrace::__rust_begin_short_backtrace::h8ad8da81466cd9cf
  24:     0x5626ff438938 - core::ops::function::FnOnce::call_once{{vtable.shim}}::hb9854ca961872107
  25:     0x5626ff3fa28f - <alloc::boxed::Box<F> as core::ops::function::FnOnce<A>>::call_once::h483711add4ba2330
                               at /rustc/4560ea788cb760f0a34127156c78e2552949f734/src/liballoc/boxed.rs:922
  26:     0x5626ff3fbb3c - <alloc::boxed::Box<F> as core::ops::function::FnOnce<A>>::call_once::h7605b45eb29ed0be
                               at /rustc/4560ea788cb760f0a34127156c78e2552949f734/src/liballoc/boxed.rs:922
  27:     0x5626ff3fbb3c - std::sys_common::thread::start_thread::h557b0c2dc4449562
                               at src/libstd/sys_common/thread.rs:13
  28:     0x5626ff3fbb3c - std::sys::unix::thread::Thread::new::thread_start::h7c2a7f9b68fe4bba
                               at src/libstd/sys/unix/thread.rs:79
  29:     0x7fc8223ae6ba - start_thread
  30:     0x7fc821ecc41d - clone
  31:                0x0 - <unknown>
thread 'block1' panicked at 'invalid key', /cargo/registry/src/github.com-1ecc6299db9ec823/slab-0.4.2/src/lib.rs:789:18
```

**Comment by bobdobs** [comment]
This could be related to issue #1327

**Comment by mzabaluev** [comment]
Looks like another manifestation of race conditions causing #1327. 
#1346 should take care of this.

**Comment by rinor** [comment]
@bobdobs I think this was fixed. Can you check and if that is the case, can we close this issue please. Thank you.

**Comment by rinor** [comment]
Closing this one since it was fixed in v0.8.5.
